lib/ui.rb

```diff
@@ -0,0 +1,142 @@
+class DictionaryUI 
+
+  #handles user interaction loop
+  def initialize
+    @loader = DictionaryLoader.new
+  end
+
+  def run
+    prompt_for_path
+    provide_statistics
+    search? ? conduct_search : goodbye
+    save_results? ? save : goodbye
+    goodbye
+  end
+
+  def prompt_for_path
+    puts puts
+    puts "Where is your dictionary? ('q' to quit)"
+    print "> "
+    path = gets.chomp
+    exit if path == 'q'
+    @dictionary = Dictionary.new(@loader.load(path))
+    puts "Dictionary was successfully loaded"
+  end
+
+  def provide_statistics
+    puts puts
+    puts "Your dictionary contains #{number_of_words} words."
+    puts "Word frequency by starting letter:"
+    words_by_starting_letter
+  end
+
+  def number_of_words(letter=0)
+    return @dictionary.processed.size if letter == 0
+    number = 0
+    @dictionary.processed.each { |word| number += 1 if word.start_with?(letter)} 
+    number
+  end
+
+  def words_by_starting_letter
+    ("A".."Z").each do |letter|
+      puts "#{letter}: #{number_of_words(letter.downcase)}"
+    end
+  end
+
+  def search?
+    puts puts
+    puts "Would you like to search your dictionary? (y/n)"
+    print "> "
+    response = gets.chomp.strip.downcase
+    until %(y n).include?(response)
+      puts "Please enter 'y' for yes and 'n' for no."
+      response = gets.chomp.strip.downcase
+    end
+    response == 'y' ? true : false
```

Comment:
Ternary operator is not required here. `response == 'y'` is the same as `response == 'y' ? true : false`